### ISSUE TITLE
Update to Polymer 2.x (planning)

### ISSUE THREAD

**Issue by fresnizky** [jtebert/jekyll-polymer#10]
Now that Polymer 2.x has been released, do you have a plan to update your theme to use Polymer 2.x?

**Comment by jtebert** [comment]
Eventually I plan to get around to this, when I have the time. If you'd like to help with this process, that would be a great help.

**Comment by samcarecho** [comment]
@jtebert how's the migration to Polymer 2.x?
Any started work? If not, can I take over the migration?

**Comment by jtebert** [comment]
I haven't had a chance to get to it. If you are interested in taking the lead on this migration, that would be greatly appreciated! I can probably help out in with parts as needed if you get the process going.

**Comment by samcarecho** [comment]
@jtebert Before migrating, I would like to plan some structural changes to the project. 
I would like to make it GitHub buildable, in other words, once pushed I would like for GitHub to build it automatically. 

I also would like to make it easy to change themes/templates.

What're your feelings about this?

**Comment by jtebert** [comment]
Making it GitHub buildable (continuous integration) seems like a somewhat independent issue from migrating to 2.0, though also something potentially worth pursuing.

I'm not sure what you mean about changing themes/templates.

**Comment by samcarecho** [comment]
@jtebert the continuous integration is kind of related as I will have to change the files structure of the project, therefore I can do it simultaneously with the migration to Polymer 2.X.
Better to plan now and already do the necessary changes, based on Polymer 2.X. then to do it later, after the project is migrated to 2.X, and risk having to do repeated work. 
We can even plan to not use gulp, but Polymer CLI. I really think it's worth at least an analysis.

By theming, I mean making it easy to change the design/layout of the resulting app.
You see, Polymer is modular, therefore we can take advantage of this fact and make the app design and layout easy to change. But this can be part of another task.

**Comment by samcarecho** [comment]
Hi @jtebert, I was waiting for a feedback from you regarding my last post.
You can assign  the update to Polymer 2.x to me.
I'm starting it today.

Is it ok if I just update your code without any structural changes?

**Comment by jtebert** [comment]
Sorry, didn't realize that you were waiting on me.

That sounds fine to me, though if you have suggestions for structural updates I'm open to that as well. Let's start a branch for it that we can merge in when it's updated and functional.

**Comment by samcarecho** [comment]
Sure. :)

While planning the update, after reading again( it's been almost 6months) part of the code, I noticed that a few structural changes will be needed, including dropping some dependencies.

For example, fontawesome-iconset is using Polymer 1.0. My suggestion in this case is getting rid of all 3rd party related to Polymer and that are not maintained by the Polymer team. Otherwise, we might end up dependent on an abandoned code, that might break jekyll-polymer.

I will try to update your code with the minimal required changes, so it's done faster.
After it's updated I recommend creating new tickets with the goal of bringing jekyll-polymer up to date with the latest practices, as the use of the Polymer CLI for building and serving.

Please, provide a feedback. :)

**Comment by samcarecho** [comment]
@jtebert as I go deeper into the update, I see less and less reasons to use Jekyll with Polymer.

I found your project by searching for a solution that would allow me to commit markdown files and this way feed with content a site or blog based on Polymer without relying on a DB.
Nonetheless, the more I think about using Jekyll with Polymer, the more I'm prone to just get rid of Jekyll and its stack (ruby, gems, etc). It seems to me that the best approach would be to just build a simple generator using Javascript, Node and Polymer, that we could seamless integrate to any Polymer app and the Polymer CLI.

What are your thoughts on this?
What are your reasons to use Jekyll? (I mean, what Jekyll features do you need?)

**Comment by samcarecho** [comment]
@jtebert I have been quite busy lately, but I have managed to rethink over the migration to Polymer 2.x.

On May, Polymer 3.0 was officially released. We should probably jump strait to 3.0.
I will spend a couple of hours checking which of the components used by your project have been upgraded to work with Polymer 3.0 and the effort needed to jump 2.0.

**Comment by samcarecho** [comment]
@jtebert, I have been trying to get back to the project but free time has been quite scarce.
If anyone with free time shows interest towards keeping the project going, please let she/he take over.

I can make myself available to help as a consultant, if you guys need it, but not as a programmer (too time demanding).

Sorry about my free time issue, and thank you for your trust, patience and understanding.
